privacyidea/lib/policy.py

```diff
@@ -758,9 +759,11 @@ def ui_get_rights(self, scope, realm, username, client=None):
         elif scope == SCOPE.USER:
             userrealm = realm
             logged_in_user["role"] = ROLE.USER
+            resolver = User(username, userrealm).resolver
```

Comment:
I was thinking: The ``User(...).resolver`` line results in a query of the Id Resolver. ``ui_get_rights`` is ultimately called by ``get_auth_token``, and there, we already have a user object:
https://github.com/privacyidea/privacyidea/blob/5e43b23f40cbf304a6246d05589fe3f4d1b280b7/privacyidea/api/auth.py#L255
So there isn't really a need to query the Id Resolver again. We could just add the ``resolver`` parameter to the ``ui_get_rights`` (and ``ui_get_main_menus``) to avoid the additional call.

On the other hand, this code is only called once at each user login, so the impact of the additional calls might be small, so we could also just keep it as above. What do you think?

Reply:
You are right. But I am not quite sure.
We only have a resolver, if a user from a resolver authenticates. We do not have  a resolver, if an internal admin authenticates! 
This way to ui_get_rights looks more self contained to me.